setup.py

```diff
@@ -73,56 +75,74 @@ def _gen_grpc(self):
-        build_dir = staging_dir / 'azure_functions_worker' / 'protos'
-        built_protos_dir = root / 'build' / 'built_protos'
-        built_proto_files_dir = (built_protos_dir
-                                 / 'azure_functions_worker' / 'protos')
+        built_protos_dir = build_dir / 'built_protos'
 
-        if os.path.exists(build_dir):
+        if os.path.exists(staging_root_dir):
             shutil.rmtree(build_dir)
 
         if os.path.exists(built_protos_dir):
             shutil.rmtree(built_protos_dir)
 
-        proto_files = glob.glob(str(proto_src_dir / '**' / '*.proto'),
-                                recursive=True)
+        shutil.copytree(proto_src_dir, staging_dir)
 
-        os.makedirs(build_dir)
-        for proto_file in proto_files:
-            shutil.copy(proto_file, build_dir)
+        os.makedirs(built_protos_dir)
 
-        protos = [os.path.basename(proto_file) for proto_file in proto_files]
+        protos = [
+            os.sep.join(('shared', 'NullableTypes.proto')),
+            os.sep.join(('identity', 'ClaimsIdentityRpc.proto')),
+            'FunctionRpc.proto'
+        ]
 
-        full_protos = []
         for proto in protos:
-            full_proto = os.sep.join(
-                ('azure_functions_worker', 'protos',
-                 'azure_functions_worker', 'protos', proto)
-            )
-            full_protos.append(full_proto)
-
-        os.makedirs(built_protos_dir)
-        subprocess.run([
-            sys.executable, '-m', 'grpc_tools.protoc',
-            '-I', os.sep.join(('azure_functions_worker', 'protos')),
-            '--python_out', str(built_protos_dir),
-            '--grpc_python_out', str(built_protos_dir),
-            *full_protos
-        ], check=True, stdout=sys.stdout, stderr=sys.stderr,
-            cwd=staging_root_dir)
-
-        compiled = glob.glob(str(built_proto_files_dir / '*.py'))
-
-        if not compiled:
+            subprocess.run([
+                sys.executable, '-m', 'grpc_tools.protoc',
+                '-I', os.sep.join(('azure_functions_worker', 'protos')),
+                '--python_out', str(built_protos_dir),
+                '--grpc_python_out', str(built_protos_dir),
+                os.sep.join(('azure_functions_worker', 'protos', proto)),
+            ], check=True, stdout=sys.stdout, stderr=sys.stderr,
+                cwd=staging_root_dir)
+
+        compiled_files = glob.glob(
+            str(built_protos_dir / '**' / '*.py'),
+            recursive=True)
+
+        if not compiled_files:
             print('grpc_tools.protoc produced no Python files',
                   file=sys.stderr)
             sys.exit(1)
 
-        for f in compiled:
-            shutil.copy(f, proto_root_dir)
+        # Needed to support absolute imports in files. See
+        # https://github.com/protocolbuffers/protobuf/issues/1491
+        self.make_absolute_imports(compiled_files)
+
+        dir_util.copy_tree(built_protos_dir, str(proto_root_dir))
```

Comment:
Do we still need **built_protos_dir**? I think we should clean it up after we build it.